tdms/include/numerical_derivative.h

```diff
@@ -1,5 +1,50 @@
+/**
+ * @file numerical_derivative.h
+ * @brief Functions to calculate the numerical derivatives.
+ *
+ * This code is the main dependency on FFTW3.
+ */
+#pragma once
 #include <fftw3.h>
-void complex_mult_vec( fftw_complex *a, fftw_complex *b, fftw_complex *c, int len);
-void init_diff_shift_op( double delta, fftw_complex *Dk, int N);
-void first_derivative( fftw_complex *in_pb_pf, fftw_complex *out_pb_pf,
-		       fftw_complex *Dk, int N, fftw_plan pf, fftw_plan pb);
+
+/**
+ * @brief Multiply two arrays of complex numbers element-wise.
+ *
+ * Performs the complex multiplication of every element of a with every element
+ * of b. Stores the result in c. All arrays must be of equal length.  Used
+ * internally by first_derivative.
```

Comment:
```suggestion
 * Performs element-wise complex multiplication of the array a with the array b.
 * a and b must be must be of equal length.
 * Stores the result in c.
 * Internally used by first_derivative.
```
Just felt the "complex multiplication of every element of a with every element of b" implied some kind of outer product was happening.

Reply:
Ah. You're right. What I _wrote_ is "every element by every element" which would mean some kind of O(n²) operation but we're just mapping complex multiplication across a list.